src/plugins/map/MQ2MapCommands.cpp

```diff
@@ -1364,6 +1580,387 @@ static void DrawMapSettings_Colors()
+		{
+			ImGui::BeginDisabled();
+			disable = true;
+		}
+		if (ImGui::Button("Reset Overrides to Defaults"))
+		{
+			ResetMapLocOverrides();
+		}
+		if (disable)
+		{
+			ImGui::EndDisabled();
+			disable = false;
+		}
+
+		ImGui::SetNextItemWidth(40);
+		ImGui::DragFloat(" Line Size (10-200, default: 50)", &gOverrideMapLocParams.lineSize, 1.0f, 10, 200, "%.0f", ImGuiSliderFlags_None);
+
+		ImGui::SetNextItemWidth(40);
+		ImGui::DragFloat(" Line Width (1-10, default: 10)", &gOverrideMapLocParams.width, 1.0f, 1, 10, "%.0f", ImGuiSliderFlags_None);
+
+		ImColor mapLocColor = gOverrideMapLocParams.color.ToImColor();
+		ImGui::SetNextItemWidth(200);
+		if (ImGui::ColorEdit3("MapLoc Override", &mapLocColor.Value.x))
+		{
+			gOverrideMapLocParams.color.Blue = static_cast<uint8_t>(mapLocColor.Value.z * 255);
+			gOverrideMapLocParams.color.Green = static_cast<uint8_t>(mapLocColor.Value.y * 255);
+			gOverrideMapLocParams.color.Red = static_cast<uint8_t>(mapLocColor.Value.x * 255);
+			gOverrideMapLocParams.color.Alpha = 255;
+			regenerate = true;
+		}
+
+		ImGui::SetNextItemWidth(40);
+		if (ImGui::DragFloat(" Radius", &gOverrideMapLocParams.circleRadius, 1.0f, 0, 1000, "%.0f", ImGuiSliderFlags_None))
+		{
+			UpdateDefaultMapLocInstances();
+		}
+
+		if (gOverrideMapLocParams.circleRadius == 0)
+		{
+			ImGui::BeginDisabled();
+			disable = true;
+		}
+		ImGui::SameLine();
+		ImColor radiusColor = gOverrideMapLocParams.circleColor.ToImColor();
+		ImGui::SetNextItemWidth(200);
+		if (ImGui::ColorEdit3("MapLoc Radius Override", &radiusColor.Value.x))
+		{
+			gOverrideMapLocParams.circleColor.Blue = static_cast<uint8_t>(radiusColor.Value.z * 255);
+			gOverrideMapLocParams.circleColor.Green = static_cast<uint8_t>(radiusColor.Value.y * 255);
+			gOverrideMapLocParams.circleColor.Red = static_cast<uint8_t>(radiusColor.Value.x * 255);
+			gOverrideMapLocParams.circleColor.Alpha = 255;
+			regenerate = true;
+		}
+		if (disable)
+		{
+			ImGui::EndDisabled();
+			disable = false;
+		}
+
+		ImGui::Unindent();
+	}
+
+	ImGui::Separator();
+
+	if (ImGui::Button("Add Map Loc"))
+	{
+		std::stringstream commandStream;
+		commandStream << addMapLocY << " " << addMapLocX << " " << addMapLocZ;
+		if (IsMapLocsStyleOverridden())
+		{
+			commandStream << " size " << gOverrideMapLocParams.lineSize
+				<< " width " << gOverrideMapLocParams.width
+				<< " color " << std::to_string(gOverrideMapLocParams.color.Red) << " " << std::to_string(gOverrideMapLocParams.color.Green) << " " << std::to_string(gOverrideMapLocParams.color.Blue)
+				<< " radius " << gOverrideMapLocParams.circleRadius
+				<< " rcolor " << std::to_string(gOverrideMapLocParams.circleColor.Red) << " " << std::to_string(gOverrideMapLocParams.circleColor.Green) << " " << std::to_string(gOverrideMapLocParams.circleColor.Blue);
+		}
+		if (addMapLocLabel[0] != '\0')
+		{
+			commandStream << " label " << addMapLocLabel;
+		}
+		char* commandStr = _strdup(commandStream.str().c_str());
```

Comment:
another _strdup

Reply:
my workarounds for getting some for of easily concat-able string into a non-const char* has been met with turmoil haha.  Is there a clean way to achieve this before I do a ton of conversions to make it work without _strdup?

Reply:
no, you need to change MapSetLocationCmd to take const char*

Reply:
This is a little weird because MapSetLocationCmd is a registered plugin command.
`void AddCommand(const char* Command, fEQCommand Function`
an fEQCommand requires non-const. 
I could maybe workaround this by creating a passthrough method just to register as an fEQCommand and then call the current method with the const signature?

Reply:
`using fEQCommand             = void   (*)(SPAWNINFO*, char* Buffer);`

Reply:
hmm i thought we had an overload for const char* too..

Reply:
you'll have to do this jank to get a char* from a string then: `&string[0]`